ui/server/src/main/scala/pl/touk/nussknacker/ui/api/ProcessesResources.scala

```diff
@@ -79,11 +80,25 @@ class ProcessesResources(val processRepository: FetchingProcessRepository,
           }
         }  ~ path("processes") {
           get {
-            complete {
-              processRepository.fetchProcesses().toBasicProcess
+            parameters(
+              'isSubprocess.as[Boolean].?,
+              'isArchived.as[Boolean].?,
+              'isDeployed.as[Boolean].?,
+              'categories.as(CsvSeq[String]).?,
+              'processingTypes.as(CsvSeq[String]).?
+            ) { (isSubprocess, isArchived, isDeployed, categories, processingTypes) =>
+              complete {
+                processRepository.fetchProcesses(
+                  isSubprocess,
+                  isArchived.orElse(Option(false)),
```

Comment:
Do we need orElse here?

Reply:
It's an back compability - there should by comment.